### DISCUSSION TITLE
Python bindings to enable standalone physics for RL

### DISCUSSION THREAD

**Discussion by jrtcppv** [microsoft/AirSim#3584] [Ideas]
I am working on a reinforcement learning application in which I only need the kinematics during training and will use the rest of AirSim (rendering, Unreal, etc) during evaluation/demonstration of the trained agents. To maximize the speed of the simulation, I was considering writing Python bindings directly on top of the C++ code using something like Boost::Python, such that something like the example [here](https://github.com/microsoft/AirSim/blob/master/Examples/StandAlonePhysics.hpp) could be achieved in Python, specifically in an OpenAI Gym environment. I understand from [this issue](https://github.com/microsoft/AirSim/issues/551) that the normal python module is not available when Unreal is disabled.

My questions are:
1) Is this a good approach for maximizing execution speed given my need for kinematics only?
2) Has this already been done?

Many thanks for any feedback!

**Comment by jonyMarino** [comment]
Hi @jrtcppv! 
You can run AirSim in headless mode. Take a look at this comment: https://github.com/microsoft/AirSim/issues/1224#issuecomment-481774139
To do a reproduction, it will be easier when we have the ExternalPhysicsEngine ready. 

Keep this discussion updated with your progress.
Good luck!

**Comment by jrtcppv** [comment]
Thank you for the response! I might try it out to compare with running the headless server. If I come up with anything I will post benchmarks here.
